tests/cargo-kani/vecdeque-cve/README.md

```diff
@@ -0,0 +1,120 @@
+This example accompanies Kani's post on CVE-2018-1000657 of the Rust Standard Library.
+
+## Files in `src`
+
+The file `issue44800.rs` is a standalone example that can be used to reproduce the CVE.
+
+The files `cve.rs` and `fixed.rs` are implementations of `VecDeque` using `raw_vec.rs` as the underlying buffer. These are taken as-is from the Standard Library except that `cve.rs` contains the issue (try a `diff` between the two). These are used in `harness.rs` in two separate modules:
+
+  - Harnesses in `cve_proofs` use the `VecDeque` implementation *with* the CVE
+  - Harnesses in `fixed_proofs` use the fixed `VecDeque` implementation
+
+Finally the file `abstract_vecdeque.rs` is a standalone abstraction of the `VecDeque` data structure that we use to prove that the `reserve` and `remove` methods maintain the data structure's resource invariant.
+
+## Dependencies
+
+  - Rust edition 2018
+  - [Kani](https://model-checking.github.io/kani/getting-started.html)
+  - [Valgrind](https://valgrind.org/) (should be run on a Linux platform)
+
+## Reproducing CVE-2018-1000657
+
+The following will rollback to an old version of the Rust toolchain, compile the example and run it using `valgrind`, a binary instrumentation framework that can detect memory issues.
+
+```bash
```

Comment:
Is there any way to do this in a sandbox so they're not messing with their main system install of Rust?

Reply:
The override is just for the directory. And we switch it back at the end of this section.

https://rust-lang.github.io/rustup/overrides.html#directory-overrides